proposed/2023/nuget-package-search.md

```diff
@@ -0,0 +1,74 @@
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+This specification will discuss three key improvements to the NuGet package search functionality. These improvements are: adding an `-ExactMatch` option to the `search` command, deprecating the `list` command in favor of `search`, and introducing a `dotnet nuget search` command to the Dotnet CLI.
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Improving the nuget search functionality across all platforms is highly requested by customers. As shown in this [issue](https://github.com/NuGet/Home/issues/5138), customers would like to have `-ExactMatch` option added to the search command. In this other [issue](https://github.com/NuGet/Home/issues/6060), we can see that it has been repeatedly requested by customers for the Dotnet CLI to adopt a search functionality. Lastly, as discussed in this [issue](https://github.com/NuGet/Home/issues/7912) the `nuget list` command is redundant with `nuget search` as both commands basically display a list of packages from a source. The only difference between the two is that the search command allows for filters. In addition, the `nuget list` command can be confusing as there is a `nuget list` command in the Dotnet CLI with a different functionality: lists configured sources and client certificates. Addressing these issues will provide users with a more straight-forward search functionality on both the Dotnet CLI and nuget.exe.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+#### 1. Adding `-ExactMatch` Option: 
+A new flag, `-ExactMatch`, will be added to the `search` command. When this flag is used, it will only return packages whose names exactly match the query.
+- **Version not specified** : In this case, the latest version of the package specified is listed .\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> --exact-match`
+- **Version specified** : In this case, the package with the specific version number is listed.\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch -Version 1.2.3.4` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> --exact-match --version 1.2.3.4`
+- **All Versions** : In this case, all versions of the package with the exact match are listed.\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch -AllVersions` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> --exact-match --all-versions`
+
+If no exact match is found, display an error: `Error: An exact match for package <PackageName> is not found.`
+#### 2. Deprecating list Command: 
```

Comment:
The `list` command appears to support `-IncludeDelisted` and `-AllVersions`. I see `-AllVersions` getting pulled forward into `search` but not `delisted`. Is that by design?

By the way, the `list -AllVersions` option has not worked for years on NuGet.org, it's tracked here: https://github.com/NuGet/NuGetGallery/issues/3274. This may mean you have an opportunity to redefine the behavior with less back-compat impact. Or it means we're finally fixing a longstanding bug :)

Reply:
I think we should try to go with a slim version of search, at least in first proposal. 
We can add more options based on community feedback while getting this design reviewed or after we ship it.